src/app/core/testing/utils/run-on-push.utils.ts

```diff
@@ -0,0 +1,12 @@
+import { ChangeDetectorRef } from '@angular/core';
+import { ComponentFixture } from '@angular/core/testing';
+
+/**
+ * Changes in components using OnPush strategy are only applied once when calling .detectChanges(),
+ * This function solves this issue.
+ */
+export async function runOnPushChangeDetection<T>(fixture: ComponentFixture<T>): Promise<void> {
+  const changeDetectorRef = fixture.debugElement.injector.get<ChangeDetectorRef>(ChangeDetectorRef);
+  changeDetectorRef.detectChanges();
```

Comment:
I'm not exactly sure what's the difference between `spectator.detectChanges()` and `spectator.detectComponentChanges()` is, but if you replace `runOnPushChangeDetection` with `spectator.detectComponentChanges()` it works.